### ISSUE TITLE
text_global_pool(pool_type='eos'): silent fallback to index 0 on missing EOS

### ISSUE THREAD

**Issue by shaun0927** [mlfoundations/open_clip#1151]
## Problem

`text_global_pool` in `src/open_clip/transformer.py`:

https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/transformer.py#L935-L940

```python
elif pool_type == 'eos':
    assert text is not None
    assert eos_token_id is not None
    idx = (text == eos_token_id).int().argmax(dim=-1)
    pooled = x[torch.arange(x.shape[0], device=x.device), idx]
```

When a sequence does not contain `eos_token_id` (truncated-before-EOS, malformed tokenizer output, caller error), `(text == eos_token_id).int()` is all-zeros and `argmax(dim=-1)` returns `0` — the function silently pools position 0 (BOS/CLS) instead of EOS. There is no warning or assertion. The v3.0.0 MetaCLIP2 WorldWide configs (`ViT-{H,L,bigG}-14-worldwide*`) all set `pool_type: "eos"`, so this affects any deployment of those models on long multilingual inputs that get truncated past the EOS.

## Repro

```python
import torch
from open_clip.transformer import text_global_pool

x = torch.arange(2 * 8 * 4, dtype=torch.float32).reshape(2, 8, 4)
text = torch.tensor([
    [1, 2, 3, 49407, 0, 0, 0, 0],   # has EOS at position 3
    [1, 2, 3, 4, 5, 6, 7, 8],       # no EOS anywhere
])
pooled = text_global_pool(x, text, pool_type='eos', eos_token_id=49407)
print(pooled[1])  # prints [32., 33., 34., 35.]  == x[1, 0], the BOS position
```

## Proposed fix

Fall back to the last-token position rather than position-0 when EOS is missing (preserves behaviour for valid inputs, avoids pooling BOS for truncated ones):

```python
elif pool_type == 'eos':
    assert text is not None
    assert eos_token_id is not None
    mask = (text == eos_token_id)
    has_eos = mask.any(dim=-1)
    idx = torch.where(
        has_eos,
        mask.int().argmax(dim=-1),
        torch.full((text.size(0),), text.size(1) - 1,
                   dtype=torch.long, device=text.device),
    )
    pooled = x[torch.arange(x.shape[0], device=x.device), idx]
```

Alternative, if you prefer strict: `assert has_eos.all()`. That is a breaking change for truncation scenarios so I would lean toward the `torch.where` form.

I will open a PR with the `torch.where` variant and a small unit test.

**Comment by rwightman** [comment]
Doesn't really matter imho, if the EOS is absent the last position absent the EOS token is not going to yield reasonable performance so don't see the value in adding code complexity.
